File: TwoPointers/LinkedList/Leetcode23.java
public class Leetcode23 {
    public ListNode mergeKLists(ListNode[] lists) {
        if(lists == null || lists.length == 0)return null;
        PriorityQueue<ListNode> q = new PriorityQueue<>((a,b)->a.val-b.val);
        for(ListNode n:lists){
            if(n != null)q.offer(n);     
        }
        ListNode dummy = new ListNode();
        ListNode curr = dummy;
        while(!q.isEmpty()){
            curr.next = q.poll();
            curr = curr.next;
            if(curr.next != null)q.offer(curr.next);
        }
        return dummy.next;
        
    }
}

File: LinkedList/Python/Leetcode23.py
import heapq
class Leetcode23:
    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
        dummy = head = ListNode()
        heap = []
        for i in range(len(lists)):
            if lists[i]:
                heapq.heappush(heap, [lists[i].val, i])
        while len(heap) > 0:
            val, i = heapq.heappop(heap)
            head.next = lists[i]
            if lists[i].next:
                lists[i] = lists[i].next
                heapq.heappush(heap, [lists[i].val, i])
            head = head.next
        return dummy.next
